app/src/main/java/de/xikolo/controllers/helper/ShortcutHelper.kt

```diff
@@ -0,0 +1,61 @@
+package de.xikolo.controllers.helper
+
+import android.annotation.TargetApi
+import android.content.Context
+import android.content.Intent
+import android.content.pm.ShortcutInfo
+import android.content.pm.ShortcutManager
+import android.graphics.drawable.Icon
+import android.os.Build
+import androidx.core.content.ContextCompat
+import androidx.core.content.ContextCompat.getSystemService
+import de.xikolo.R
+import de.xikolo.controllers.course.CourseActivityAutoBundle
+import de.xikolo.storages.RecentCoursesStorage
+
+const val MAX_SHORTCUTS = 4
+
+class ShortcutHelper {
+
+    @TargetApi(Build.VERSION_CODES.N_MR1)
+    fun addCourse(context: Context, courseId: String, title: String) {
+
+        val recentCoursesStorage = RecentCoursesStorage()
+        recentCoursesStorage.addCourse(courseId, title)
+
+        updateShortcuts(context)
+    }
+
+    @TargetApi(Build.VERSION_CODES.N_MR1)
+    fun updateShortcuts(context: Context) {
+
+        val shortcutManager = getSystemService<ShortcutManager>(context, ShortcutManager::class.java)
+        val recentCourses = RecentCoursesStorage().recentCourses
+        val intentList = mutableListOf<ShortcutInfo>()
+
+        for (course in recentCourses) {
+            val courseId = course.first
+            val title = course.second
+
+            val intent = CourseActivityAutoBundle.builder().courseId(courseId).build(context)
+            intent.action = Intent.ACTION_VIEW
+
+            val shortcut = ShortcutInfo.Builder(context, courseId)
+                .setShortLabel(title)
+                .setIcon(Icon.createWithResource(context, R.drawable.ic_shortcut_course))
```

Comment:
I guess we could use a vector graphic instead of PNGs here like we do it now for other icons as well?